internal/compose/compose.go

```diff
@@ -340,7 +345,29 @@ func (p *Project) runDockerComposeCmd(opts dockerComposeOptions) error {
 	return cmd.Run()
 }
 
+func (p *Project) DockerComposeVersion() (string, error) {
```

Comment:
As this method is used only by `ContainerName`, do you think we could make it package-visible and access it only in `ContainerName`?

This way you would prevent adjusting all calling code. If we need this condition in other places, we can refactor it then.

Reply:
Sounds good. I have refactored this to make it package-visible. To avoid calling `docker-compose --version` each time `ContainerName` is invoked I've stored the value in a project variable.